addons/partner_autocomplete/static/src/js/partner_autocomplete_fieldchar.js

```diff
@@ -21,9 +21,8 @@ export class PartnerAutoCompleteCharField extends CharField {
 
     async validateSearchTerm(request) {
         if (this.props.name == 'vat') {
-            return this.partner_autocomplete.isVATNumber(request);
-        }
-        else {
+            return await this.partner_autocomplete.isVATNumber(request) || this.partner_autocomplete.checkGSTNumber(request);
```

Comment:
Could you create a new function `isTaxNumber` ? There are already two occurrences where we check if the value is either a VAT number or a GST number.
By the way, could you also rename `checkGSTNumber` to `isGTSNumber` for consistency ?